docs/16-jak-dziala-antyspam.md

```diff
@@ -0,0 +1,59 @@
+Dlatego przy tym mechanizmie mamy sporo wartości konfiguracyjnych - takich jak wymagana średnia dzienna ilość wiadomości w ostatnim czasie, czas bycia na serwerze, minimalna suma wszystkich wiadomości i inne.  
+Wszystko to, aby zminimalizować ryzyko przypięcia tej etykiety do użytkownika, który nie jest aktywny regularnie (z naciskiem na regularnie), przy czym jednocześnie nie sprawia problemów.
+
+## Czym są detektory
+
+Powiedzieliśmy sobie o tym jak dostosowujemy kary do różnych typów użytkowników, teraz pora omówić skąd te kary w ogóle się biorą.
+
+Detektory zajmują się rozpoznawaniem konkretnych specyficznych zagrożeń i zwracaniem informacji jaka jest szansa na to zagrożenie - przykładowo zerowa, niska, średnia... lub że są pewne.  
+Na podstawie tego jak dużo detektorów w jakim stopniu jest pewnych tego, za co są odpowiedzialne, system decyduje o ostatecznej decyzji.  
+Po otrzymaniu podsumowania wyników detektorów, ustalamy kare dla użytkownika na podstawie wspomnianego podsumowania i wspomnianego typu użytkownika (niebezpieczny/neutralny/bezpieczny).
+
+Przyjęliśmy zasadę że zawsze pierwszym krokiem jest ostrzeżenie o wykrytym potencjalnym spamie, które nie jest niebezpieczne - jednak jeśli ktoś zignoruje ostrzeżenie, zostaje wyciszony.  
+Czas wyciszenia jest zależny od tego jak dużo razy został wyciszony w ostatnim czasie.
+
+## Lista domyślnych detektorów
+
+#### LinksDetectorStrategy
+
+Sprawdza czy wiadomość zawiera link - chcemy aby takie wiadomości zwiększały szanse na bycie wykrytymi jako spam, żeby ewentualny spam groźnymi linkami został wykryty możliwie najwcześniej.
+
+#### DuplicatedMessagesDetectorStrategy
+
+Porównuje aktualną wiadomość z poprzednimi - jeśli ktoś wysłał wiele razy tą samą wiadomość, istnieje duże prawdopodobieństwo że jest to nie dość że spam, to jeszcze bezmyślny.  
+Wewnątrz tego detektora działa również algorytm porównywania wiadomości, więc nawet jeśli ktoś nie wysyła identycznych tekstów, ale są one bardzo podobne - zostaną rozpoznane jako duplikaty, z tym że nie z 100% prawdopodobieństwem.
+
+#### CapslockDetectorStrategy
+
+Osoby które nie dość że spamują, to w dodatku robią to z caps lockiem, powinny być szybciej wykrywane.  
+Powód - dla wielu osób wiadomości pisane głównie caps lockiem są irytujące.  
+Im szybciej ktoś przestanie denerwować innych, tym lepiej.
+
+#### FloodDetectorStrategy
+
+Detektor będący tym, co wiele osób rozumie jako system anty spamowy - czyli sprawdzanie jak dużo wiadomości zostało wysłanych w określonym czasie.
+
+Dodatkowo część detektorów jest ustawionych na sprawdzanie również wiadomości innych użytkowników - ma to na celu chronienie serwera przed sytuacją, gdzie nie atakuje nas jeden bot/użytkownik spamujący, tylko jest ich bardzo dużo, a każdy z nich wysyła wiadomość w dużych odstępach czasu.
```

Comment:
w sumie to jeszcze żaden detector tego nie robi, ale można by dać np. duplicateDetector aby brał pod uwagę wszystkich użytkowników

Reply:
kurde, to miało być w podstawowych założeniach

Reply:
znaczy poprawka tego zajmie koło 2 minut i potem jeszcze tylko kwestia dopisania testów

Reply:
możesz zrobić taska i oznaczyć jako FastTask